packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -16,14 +16,19 @@ import { isCompanionActivated } from '../lib/element';
 import { AuthTriggers, AuthTriggersOrString } from '../lib/auth';
 import { Squad } from '../graphql/squads';
 
-export type LoginState = { trigger: AuthTriggersOrString };
+export interface LoginState {
+  trigger: AuthTriggersOrString;
+  referral?: string;
+}
+
+type LoginOptions = Omit<LoginState, 'trigger'>;
 
 export interface AuthContextData {
   user?: LoggedUser;
   referral?: string;
   trackingId?: string;
   shouldShowLogin: boolean;
-  showLogin: (trigger: AuthTriggersOrString) => void;
+  showLogin: (trigger: AuthTriggersOrString, options?: LoginOptions) => void;
```

Comment:
Alternatively would it not be better to make this a Object? Then we can simply pass LoginState?

Reply:
There would be too many changes with the current implementations and I didn't want to put too much stress on the review. Probably let's do it in a separate ticket. WDYT?

Reply:
There would be too many changes with the current implementations and I didn't want to put too much stress on the review. Probably let's do it in a separate ticket. WDYT?

Reply:
Yeah makes sense, we can do it separate, just lodge a ticket please.